sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java

```diff
@@ -155,23 +185,34 @@ public static LocalDate parseLocalDateQuietly(@Nullable String s) {
   }
 
   /**
-   * @param s string in format {@link #DATETIME_FORMAT}
-   * @throws SonarException when string cannot be parsed
+   * @param s date time in ISO-8601 format. For historical reason, the format {@link #DATETIME_FORMAT} is supported
+   * @throws RuntimeException when string cannot be parsed
    */
   public static Date parseDateTime(String s) {
     return Date.from(parseOffsetDateTime(s).toInstant());
   }
 
   /**
-   * @param s string in format {@link #DATETIME_FORMAT}
-   * @throws SonarException when string cannot be parsed
+   * @param s date time in ISO-8601 format. For historical reason, the format {@link #DATETIME_FORMAT} is supported
+   * @throws RuntimeException when string cannot be parsed
    * @since 6.6
    */
   public static OffsetDateTime parseOffsetDateTime(String s) {
     try {
-      return OffsetDateTime.parse(s, DATETIME_FORMATTER);
+      TemporalAccessor dateTime = ISO_DATE_TIME.parseBest(s, OffsetDateTime::from, LocalDateTime::from);
+      if (dateTime instanceof OffsetDateTime) {
+        return (OffsetDateTime) dateTime;
+      } else if (dateTime instanceof LocalDateTime) {
+        return ((LocalDateTime) dateTime).atZone(ZoneId.systemDefault()).toOffsetDateTime();
+      }
+
+      throw unsupportedIsoDateTimeFormatException(s);
     } catch (DateTimeParseException e) {
-      throw MessageException.of("The date '" + s + "' does not respect format '" + DATETIME_FORMAT + "'", e);
+      try {
+        return OffsetDateTime.parse(s, DateTimeFormatter.ofPattern(DATETIME_FORMAT));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "DATETIME_FORMAT"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)